### DISCUSSION TITLE
Modelica.Fluid.Interfaces.FluidPort_a

### DISCUSSION THREAD

**Discussion by XiongbaXu** [lbl-srg/modelica-buildings#3516] [Q&A]
I developed a Modelica model, I want to link this model to other models, so I used the port, but I encounter a question, when I want to get the value of port.C_outflow, I always meet problems:Cannot resolve type of expression 1 - port_in.C_outflow. The operands have types Integer, Real[0] in component <NO_COMPONENT>.

![image](https://github.com/lbl-srg/modelica-buildings/assets/140901145/14f1b01c-ba72-4ae1-abce-352b6059dd31)
![30409b5c1871795ab5a43982b98b7cf](https://github.com/lbl-srg/modelica-buildings/assets/140901145/0ed50a35-a538-4bc1-8910-ac9cf7b54cf3)
![7d10e90e594c222a92bd21734f9e2f9](https://github.com/lbl-srg/modelica-buildings/assets/140901145/6c87616a-6f98-41dd-8561-06cac1fb142c)

**Comment by mwetter** [comment]
The error message
```
1 - port_in.C_outflow. The operands have types Integer, Real[0] 
```
likely comes from the situation that `port_in.C_outflow` is a vector of Reals of dimension 0, and that is subtracted from an integer (`1`).
You may want to look at `Buildings.Fluid.Sensors.TraceSubstances` for how to properly access trace substances.

**Reply by XiongbaXu** [reply]
Thank you. It is reasonable to use sensors, but when I tried sensors, I encountered another question.
![image](https://github.com/lbl-srg/modelica-buildings/assets/140901145/5913348a-c684-4b5a-be7e-14cf0f8e43a6)
![image](https://github.com/lbl-srg/modelica-buildings/assets/140901145/1906d6a0-ba21-4fba-9821-39a20864ac7a)
![image](https://github.com/lbl-srg/modelica-buildings/assets/140901145/cfbfdf16-06f6-44e9-89a8-4319ca7f535c)
